### DISCUSSION TITLE
Folder not writable from guest when 9p, linux, libvirt

### DISCUSSION THREAD

**Discussion by misog** [vagrant-libvirt/vagrant-libvirt#1459] [Q&A] [answered]
Hi, I installed vagrant with NIX package manager in Debian. Guest/box is generic/fedora35 (libvirt, 3.6.8)
i have this in Vagrantfile:
```
config.vm.synced_folder "./", "/vagrant", type: "9p", disabled: false, accessmode: "squash", owner: "1000"
```
It mounts the host folder to guest /vagrant correctly and I can see files from guest however I can not write to any file and I can not create any file from guest. I want to have bidirectional write access to shared folders.
```
[vagrant@fedora35 vagrant]$ ls -la
total 16
drwxr-xr-x.  3 vagrant vagrant 4096 Mar  4 13:39 .
dr-xr-xr-x. 18 root    root     239 Mar  4 12:08 ..
-rwxrwxrwx.  1 vagrant vagrant   11 Mar  4 13:38 test.txt
drwxr-xr-x.  4 vagrant vagrant 4096 Mar  4 11:27 .vagrant
-rw-r--r--.  1 vagrant vagrant 3231 Mar  4 13:19 Vagrantfile
```

```
[vagrant@fedora35 vagrant]$ echo "a" > test.txt 
-bash: test.txt: Operation not permitted
[vagrant@fedora35 vagrant]$ 
```

EDIT:

I set 777 on the folder and now I can create files - but not rewrite files from guest. Files are created with libvirt-qemu owner and group (host user).

```
config.vm.synced_folder "./", "/vagrant", type: "9p", accessmode: "squash", owner: "1000", access: "1000"
```

I tried to specify both owner: "1000" and access: "1000" but no change.

EDIT 2:

I have partial success with this:
```
config.vm.synced_folder "./", "/vagrant", type: "9p", accessmode: "mapped", owner: "1000", access: "1000"
```
Now from guest can create and rewrite files because group is correct:
```
-rw-rw-r--  1   64055 vagrant    3 Mar  4 19:42 partial.txt
```

But host is still:
```
-rw------- 1 libvirt-qemu libvirt-qemu    3 mar  4 20:42 partial.txt
```

Thank you for help

**Comment by electrofelix** [comment, marked as answer]
Suggest try using the virtiofs type instead, based on some issues in #378 it doesn't feel like this type is straight forward to use or get the desired behaviour.
